### DISCUSSION TITLE
azure-pipelines manager: Support container jobs?

### DISCUSSION THREAD

**Discussion by Tobias-08** [renovatebot/renovate#21844] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.6.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

2020

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate supports updating (root) `container`-element in [container jobs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=azure-devops)

### What you tried so far.

Renovate seems to support updating `container`-elements beneath `resources` ([service containers](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/service-containers?view=azure-devops&tabs=yaml)) but not updating [container jobs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=azure-devops). Is this the current state or am I doing something wrong?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by HonkingGoose** [comment]
Did you read the [Automated Dependency Updates for Azure Pipelines](https://docs.renovatebot.com/modules/manager/azure-pipelines/) docs? The feature is in beta, but it's not clear to me what parts we support and what still needs work. 😱 

I found this open issue, don't know if that is related:

- #15028

**Reply by Tobias-08** [reply]
Hi @HonkingGoose, thanks for your reply.

I read the docs and it wasn't clear for me, too, what parts are supported. One might interpret the docs that the feature I asked for is not yet supported (because it is not explicitly mentioned).

The mentioned open issue is not related.

**Reply by HonkingGoose** [reply]
@JamieMagee do you know if Renovate supports updating containers jobs when using the `azure-pipelines` manager?

---

> I read the docs and it wasn't clear for me, too, what parts are supported. One might interpret the docs that the feature I asked for is not yet supported (because it is not explicitly mentioned).

We should update the docs with whatever answer the maintainers give, so that our users know what's working and what's missing. 😉 

> The mentioned open issue is not related.

Thanks for checking that!

**Reply by JamieMagee** [reply, marked as answer]
@Tobias-08 is right, we don't currently support container jobs for `azure-pipelines`, only container resources and Azure Pipelines tasks. Adding support should be relatively straightforward, as we already extract container images from the container resources section of the Azure Pipelines document.

**Reply by rarkins** [reply]
Created feature request: #21987
